src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/ReadWriteSplittingPlugin.java

```diff
@@ -66,24 +76,40 @@ public ReadWriteSplittingPlugin(
       PropertySet propertySet,
       IConnectionPlugin nextPlugin,
       Log logger) {
-    this(currentConnectionProvider, new BasicConnectionProvider(), propertySet, nextPlugin, logger);
+    this(
+        currentConnectionProvider,
+        new AuroraTopologyService(logger),
+        new BasicConnectionProvider(),
+        new RdsHostUtils(logger),
+        propertySet,
+        nextPlugin,
+        logger);
   }
 
   ReadWriteSplittingPlugin(
       ICurrentConnectionProvider currentConnectionProvider,
+      ITopologyService topologyService,
       IConnectionProvider connectionProvider,
+      RdsHostUtils rdsHostUtils,
       PropertySet propertySet,
       IConnectionPlugin nextPlugin,
       Log logger) {
     this.currentConnectionProvider = currentConnectionProvider;
+    this.topologyService = topologyService;
     this.connectionProvider = connectionProvider;
+    this.rdsHostUtils = rdsHostUtils;
     this.propertySet = propertySet;
     this.nextPlugin = nextPlugin;
     this.logger = logger;
   }
 
   @Override
-  public Object execute(Class<?> methodInvokeOn, String methodName, Callable<?> executeSqlFunc, Object[] args) throws Exception {
+  public Object execute(
```

Comment:
NIT: was this line really too long?

Reply:
Probably could go either way. It's broken up like this in FailoverConnectionPlugin so I just did the same here